zcash_client_backend/src/scanning.rs

```diff
@@ -124,9 +124,21 @@ pub enum ScanError {
         new_height: BlockHeight,
     },
 
-    /// The size of the Sapling note commitment tree was not provided as part of a [`CompactBlock`]
-    /// being scanned, making it impossible to construct the nullifier for a detected note.
-    SaplingTreeSizeUnknown { at_height: BlockHeight },
+    /// The note commitment tree size for the given protocol
+    TreeSizeMismatch {
+        protocol: ShieldedProtocol,
+        at_height: BlockHeight,
+        given: u32,
+        computed: u32,
+    },
+
+    /// The size of the note commitment tree for the given protocol was not provided as part of a
+    /// [`CompactBlock`] being scanned, making it impossible to construct the nullifier for a
+    /// detected note.
+    TreeSizeUnknown {
+        protocol: ShieldedProtocol,
+        at_height: BlockHeight,
+    },
```

Comment:
Changes should be noted in the changelog. IIRC `TreeSizeUnknown` case was previously added in this set of PRs, so its entry just needs to be adjusted, but `TreeSizeMismatch` is new.

Reply:
Since the `Chain` error type was removed and restructured, I just note `ScanError` as being an entirely new addition; a user migrating from the `Chain` and `ChainError` types will need to look at `ScanError` in its entirety anyway.